lib/rules/no-global-setup.js

```diff
@@ -10,24 +10,43 @@ var setupRegexp = /^(before|after)(Each|All)$/
 
 module.exports = function (context) {
   var suiteDepth = 0
+  var hasGlobalSetup = false
+  var hasRootSuite = false
+  var globalSetupNodes = []
   return {
     CallExpression: function (node) {
       if (suiteRegexp.test(node.callee.name)) {
+        hasRootSuite = true
         suiteDepth++
       } else if (setupRegexp.test(node.callee.name) && suiteDepth === 0) {
-        context.report({
-          data: {
-            name: node.callee.name
-          },
-          message: 'Do not use `{{name}}` outside a `describe`.',
-          node
+        hasGlobalSetup = true
+        globalSetupNodes.push({
+          name: node.callee.name,
+          loc: {
+            line: node.loc.line,
+            column: node.loc.column
+          }
         })
       }
     },
     'CallExpression:exit': function (node) {
       if (suiteRegexp.test(node.callee.name)) {
         suiteDepth--
       }
+    },
+    'Program:exit': function () {
+      if (hasGlobalSetup && hasRootSuite) {
+        for (var i = 0, nodeDetail, len = globalSetupNodes.length; i < len; i++) {
+          nodeDetail = globalSetupNodes[i]
+          context.report({
+            data: {
+              name: nodeDetail.name
+            },
+            message: 'Do not use `{{name}}` outside a `describe` except for in global helpers.',
+            loc: nodeDetail.loc
+          })
+        }
+      }
     }
```

Comment:
@DianaSuvorova much better this way... Each one is properly reported separately and the location is preserved now.